# Airbub Data Analysis Project

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [4]:
air_bnb = pd.read_csv('AB_NYC_2019 - AB_NYC_2019.csv')

air_bnb.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [31]:
#Which hosts are the busiest and why?

busiest_host=air_bnb['host_name'].value_counts().sort_values(ascending=False)
print(busiest_host[0:5])

# Michael, David ,Sonder (NYC), John, Alex are the most busy hosts since they are frequently involving as it is proofed using value_counts


host_name
Michael         417
David           403
Sonder (NYC)    327
John            294
Alex            279
Name: count, dtype: int64


In [18]:
#How many neighborhood groups are available and which shows up the most?
air_bnb['neighbourhood_group'].value_counts()



neighbourhood_group
Manhattan        21661
Brooklyn         20104
Queens            5666
Bronx             1091
Staten Island      373
Name: count, dtype: int64

In [67]:
# Are private rooms the most popular in manhattan?
air_bnb.groupby(['neighbourhood_group','room_type'])['neighbourhood_group'].count()['Manhattan'] 

# The Entire hom/apt is the most popular in manhattan


room_type
Entire home/apt    13199
Private room        7982
Shared room          480
Name: neighbourhood_group, dtype: int64

In [33]:
# Which hosts are the busiest and based on their reviews?
busiest_review=air_bnb.groupby(['host_name'])['number_of_reviews'].max().sort_values(ascending=False)
print(busiest_review[0:5])

host_name
Dona        629
Jj          607
Maya        543
Carol       540
Danielle    510
Name: number_of_reviews, dtype: int64


In [55]:
#Which neighorhood group has the highest average price?
Avg_price=air_bnb.groupby(['neighbourhood_group'])['price'].agg(['mean']).sort_values(by=['mean'], ascending= False)
print(Avg_price[0:1])

#a.sort_values(by=['mean'], ascending= False)

#'Manhattan' has the highest average price

                           mean
neighbourhood_group            
Manhattan            196.875814


In [59]:
# Which neighbor hood group has the highest total price?
highest_total_price=air_bnb.groupby(['neighbourhood_group'])['price'].max().sort_values(ascending=False)
print(highest_total_price[0:3])

neighbourhood_group
Brooklyn     10000
Manhattan    10000
Queens       10000
Name: price, dtype: int64


In [19]:
#Which top 5 hosts have the highest total price?
highest_Price=air_bnb.groupby(['host_name','price'])['price'].max().sort_values(ascending=False)
#highest_Price.sort_values(ascending=False)
# highest=[]
# for i in highest_Price[i]:
#     if  highest_Price[i]<5:
#          highest_Price[i].append(highest)
#     print(highest)
print(highest_Price[0:5])

host_name  price
Jelena     10000    10000
Kathrine   10000    10000
Erin       10000    10000
Matt       9999      9999
Amy        9999      9999
Name: price, dtype: int64


In [63]:
# Who currently has no (zero) availability with a review count of 100 or more?['availability_365'== 0]['number_of_reviews' >= 0]]

AirB_host=air_bnb.set_index(['host_name'])
#A
availability=(AirB_host['availability_365'] == 0) 
Review= (AirB_host[availability]['number_of_reviews'] >= 100)
Result= AirB_host[availability][Review]

#Result
print(Result['number_of_reviews'])
print(Result['availability_365'])


host_name
MaryEllen     118
Christiana    168
Sol           193
Coral         114
Doug          206
             ... 
Kathleen      103
Janet         119
Albert        102
Stephany      131
Mariluz       112
Name: number_of_reviews, Length: 162, dtype: int64
host_name
MaryEllen     0
Christiana    0
Sol           0
Coral         0
Doug          0
             ..
Kathleen      0
Janet         0
Albert        0
Stephany      0
Mariluz       0
Name: availability_365, Length: 162, dtype: int64


In [23]:
# What host has the highest total of prices and where are they located?
#air_bnb['host_name'].max(['price'])
host_price_loc=air_bnb.groupby(['host_name','price','neighbourhood_group'])['price'].max().sort_values(ascending=False)
print(host_price_loc[0:3])

host_name  price  neighbourhood_group
Jelena     10000  Manhattan              10000
Kathrine   10000  Queens                 10000
Erin       10000  Brooklyn               10000
Name: price, dtype: int64


In [65]:
# When did Danielle from Queens last receive a review?


#air_bnb['host_name','neighbourhood_group').loc['Danielle']['Queens']


#air_bnb[air_bnb['host_name'['Danielle']]]
Dan=air_bnb.set_index(['host_name']).loc['Danielle']
Dan
Reviwe=(Dan['neighbourhood_group']== 'Queens' )
last_review_date=Dan[Reviwe]['last_review'].sort_values(ascending= False)
print(last_review_date[0:1])


#Daniel.loc[['Danielle'],['Queens','last_review']]

host_name
Danielle    2019-07-08
Name: last_review, dtype: object


## Further Questions

1. Which host has the most listings?

In [26]:
listing=air_bnb.groupby(['host_name'])['calculated_host_listings_count'].agg('max').sort_values(ascending=False)
print(listing[0:1])

host_name
Sonder (NYC)    327
Name: calculated_host_listings_count, dtype: int64


2. How many listings have completely open availability?

In [248]:
AirB_listings=air_bnb.set_index(['calculated_host_listings_count'])
#AirB_listings
Zero_Ava= (AirB_listings['availability_365'] == 0)
AirB_listings[Zero_Ava]['availability_365'].value_counts()

availability_365
0    17533
Name: count, dtype: int64

3. What room_types have the highest review numbers?

In [66]:
review_numbers=air_bnb.groupby(['room_type'])['number_of_reviews'].max().sort_values(ascending=False)
print(review_numbers[0:1])

room_type
Private room    629
Name: number_of_reviews, dtype: int64


# Final Conclusion

1. Which hosts are the busiest and why?

Michael, David ,Sonder (NYC), John, Alex are the most busy hosts since they are frequently booked by customers as it is proven using value_counts.

2. How many neighborhood groups are available and which shows up the most?

There are five neighborhood groups available , These are Manhattan, Brooklyn, Queens, Bronx, and Staten Island

3. Are private rooms the most popular in manhattan?

  The Entire hom/apt is the most popular in manhattan than private rooms

4. Which hosts are the busiest and based on their reviews?
  According to the review Dona,Jj, Maya, Carol, and Danielle are the top busiest hosts

5. Which neighorhood group has the highest average price?

  Manhattan' has the highest average price with $197 aveage price

6. Which neighbor hood group has the highest total price?
   Brookly, Manhattan and Queens are the three neighbor group with the highest total price    of $10,000

7. Which top 5 hosts have the highest total price?
  Jelena, kathrine, Erin, Matt, and Amy are the five host with highese total price

8. Who currently has no (zero) availability with a review count of 100 or more?
  These hosts are some of the 162 host whove have above 100 review count and has no availablity.
   *MaryEllen  ,Christiana  ,Sol ,Coral  ,Doug ,Kathleen ,Janet  ,Albert ,Stephany  ,Mariluz   
9. What host has the highest total of prices and where are they located?
  Jelena, kathrine, Erin has the highest total price and are located in Manhattan,Queens,     and Brooklyn consecutively  

10. When did Danielle from Queens last receive a review?
   Danielle from Queens received his last review in July 8, 2019.

Further Questions
1. Which host has the most listings 
Sonder has the highest listings
2. How many listings have completely open availability? 
17533 listings have open availabltity 
3. What room_types have the highest review numbers? 
Private room has the highest review numbers.
